### PR TITLE
Build inputs from templates, version 2

### PR DESCRIPTION
**Pull request by cwhelan**
This PR replaces https://github.com/broadinstitute/gatk-sv/pull/67 with updated functionality based on @mwalker174 's review. I'm creating a new PR because rebasing the old branch against multiple changed input files was too cumbersome.

I've attempted to address all 4 of the points made here in the latest round of changes: https://github.com/broadinstitute/gatk-sv/pull/67#pullrequestreview-517086284

Please see the comments in `scripts/inputs/build_inputs.py` for information on how this new functionality works.

The `build_inputs.py` script is now invoked by `scripts/inputs/build_default_inputs.sh`. This script builds input files into four directories:

```
inputs/
test_inputs_small/
test_inputs_large/
test_inputs_single_sample/
```

Where the files in each output directory are populated with different test batches (the old `test_small` and `test_large` batches, and the single sample input files with NA12878 + 1kg ref panel and NA19240 + ref_panel_v1b.

There are a few test input files that I couldn't really associate with the concept of a "test_batch": the files for Module08 and TrainGCNV. Outputs for these will therefore appear in all three of the `test_` outputs. We can fix this by defining their inputs in terms of properties specific to a test batch input values file.

### CODE REVIEW

**Review by mwalker174 — COMMENTED** [review]
This is great! I suspect this would be useful for other projects. Have you considered creating a separate repo for the script? I think it could be easily modified to be more general-purpose.

File: scripts/inputs/build_inputs.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import sys
+import argparse, os, os.path
+import glob
+import json
+from jinja2 import Environment, Template, FileSystemLoader, meta, Undefined
```
[the remark this answers is not in the document]
Comment (cwhelan) [reply scripts/inputs/build_inputs.py]: Removed..

File: scripts/inputs/build_default_inputs.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+scripts/inputs/build_inputs.py input_templates inputs -a '{"ref_panel" : "ref_panel_1kg_v2"}'
```
[the remark this answers is not in the document]
Comment (cwhelan) [reply scripts/inputs/build_default_inputs.sh]: Changed the scripts to take a base directory (as many of our other scripts do)

File: scripts/inputs/build_inputs.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        target_subdir = os.sep.join(split_path[0:i])
+        if not os.path.isdir(target_subdir):
+            os.mkdir(target_subdir)
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument("template_directory")
```
[the remark this answers is not in the document]
Comment (cwhelan) [reply scripts/inputs/build_inputs.py]: Implemented this feature and refactored the script into a few methods to make it easier.

**Comment by cwhelan** [comment]
@mwalker174 how does this look to you now?

**Comment by mwalker174** [comment]
Thanks for the changes, looks great! We should advertise this at methods meeting. Merge when ready.

**Pull request merged**